### PR TITLE
StepRws in EVM circuits

### PR DESCRIPTION
**Pull request by nullity00**
### Description

Similar to #1582 , We use the ``StepRws`` struct to automatically track read-write indices.

### Issue Link

#1586 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Contents

The change has been made to the following files

- [x] -  create.rs
- [x] -  end_tx.rs
- [x] -  error_oog_call.rs
- [x] -  error_oog_memory_copy.rs
- [x] -  error_oog_sload_sstore.rs
- [x] -  error_return_data_oo_bound.rs
- [x] -  error_write_protection.rs
- [x] -  extcodecopy.rs
- [x] -  extcodehash.rs
- [x] -  extcodesize.rs
- [x] -  logs.rs
- [x] -  return_revert.rs
- [x] -  returndatacopy.rs
- [x] -  sha3.rs
- [x] -  sload.rs
- [x] -  sstore.rs

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/create.rs [hunk 1]
```diff
@@ -658,11 +666,8 @@ impl<F: Field, const IS_CREATE2: bool, const S: ExecutionState> ExecutionGadget<
             } else if init_code_length.as_usize() == 0 {
                 F::ONE
             } else {
-                block
-                    .get_rws(step, 18 + rw_offset + copy_rw_increase - transfer_offset)
-                    .call_context_value()
-                    .to_scalar()
-                    .unwrap()
+                rws.offset_add(5 - transfer_offset); // rw_off + 13 + 5 + copy_rw_increase - transfer_offset
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/evm_circuit/execution/create.rs:669]: Don't think this is required.
```suggestion
```

**Review by ChihChengLiang — CHANGES_REQUESTED** [review]
Hi @nullity00,
Thanks for taking this issue! I did the first found of review and left some comments. Please take a look.

**Comment by ChihChengLiang** [comment]
Hi @nullity00, do you see any blocker in this PR? Is there anything we can help?

Note that you can use `block.debug_print_txs_steps_rw_ops();` to debug the read-writes, as #1586 describes.

**Comment by ed255** [comment]
Closing due to inactivity.  @nullity00 if you would like to retake this let us know and we can open the PR again :)

**Pull request closed without merging**